plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/CodeWhispererCodeScanSession.kt

```diff
@@ -263,10 +279,21 @@ class CodeWhispererCodeScanSession(val sessionContext: CodeScanSessionContext) {
     }
 
     @Throws(IOException::class)
-    fun uploadArtifactToS3(url: String, fileToUpload: File, requestHeaders: Map<String, String>?) {
+    fun uploadArtifactToS3(url: String, uploadId: String, fileToUpload: File, md5: String, kmsArn: String?, requestHeaders: Map<String, String>?) {
+        val uploadIdJson = """{"uploadId":"$uploadId"}"""
         HttpRequests.put(url, "application/zip").userAgent(AwsClientManager.getUserAgent()).tuner {
-            requestHeaders?.forEach { entry ->
-                it.setRequestProperty(entry.key, entry.value)
+            if (requestHeaders.isNullOrEmpty()) {
+                it.setRequestProperty(CONTENT_MD5, md5)
+                it.setRequestProperty(CONTENT_TYPE, APPLICATION_ZIP)
+                it.setRequestProperty(SERVER_SIDE_ENCRYPTION, AWS_KMS)
```

Comment:
why is this required?

Reply:
This is fallback if request headers are empty. the new QCA call takes request headers and old RTS prod takes the old info